docker/src/main/java/brooklyn/entity/container/docker/application/DockerEntitySpecResolver.java

```diff
@@ -71,16 +57,12 @@ public String getBrooklynType(String serviceType) {
         String imageName = Iterables.get(parts, 0);
         String imageTag = Iterables.get(parts, 1, "latest");
         log.debug("Creating Docker service entity with image {} and tag {}", imageName, imageTag);
+
+        EntitySpec<VanillaDockerApplication> spec = EntitySpec.create(VanillaDockerApplication.class);
         spec.configure(DockerAttributes.DOCKER_IMAGE_NAME, imageName);
         if (parts.size() == 2) {
             spec.configure(DockerAttributes.DOCKER_IMAGE_TAG, imageTag);
         }
-        if (resolver.getAttrs().containsKey("id")) {
-            String containerName = (String) resolver.getAttrs().getStringKey("id");
-            spec.configure(DockerAttributes.DOCKER_CONTAINER_NAME, containerName);
-        }
-
-        super.decorateSpec(resolver, spec);
```

Comment:
This is part of the cherry picked commit https://github.com/brooklyncentral/clocker/commit/5eaf90ff267a702c9d9da6e322d393fd953e9c1e from https://github.com/brooklyncentral/clocker/pull/178, I just merged and deployed a few blueprints to check, I didn't dig into the specifics. I'll have a closer look.